signature/openpgp/openpgp.go

```diff
@@ -0,0 +1,86 @@
+package openpgp
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"strings"
+	"time"
+
+	"github.com/containers/image/types"
+
+	"golang.org/x/crypto/openpgp"
+)
+
+type context struct {
+	keyring openpgp.EntityList
+}
+
+type OpenPGPMechanism struct {
+	ctx *context
+}
+
+func NewOpenPGPSigningMechanism() (types.SigningMechanism, error) {
+	return OpenPGPMechanism{ctx: &context{
+		keyring: openpgp.EntityList{},
+	}}, nil
+}
+
+func (m OpenPGPMechanism) ImportKeysFromBytes(blob []byte) ([]string, error) {
+	keyring, err := openpgp.ReadArmoredKeyRing(bytes.NewReader(blob))
```

Comment:
Note that this should also accept unarmored data; with `keyData` inside `policy.json` containing a base64-encoded representation of `keyData`, it would be rather silly, and wasting space, to have a base64-encoded of a base64-based armor encoding of the underlying key.